Android/app/src/main/java/uncommon/common/activity/PlaceActivity.java

```diff
@@ -1,106 +1,131 @@
+
+import com.facebook.login.LoginManager;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+import uncommon.common.R;
+import uncommon.common.api_interface.ApiInterface;
+import uncommon.common.models.ClassList;
+import uncommon.common.models.Reservation;
+import uncommon.common.network.RetrofitInstance;
+
+public class PlaceActivity extends AppCompatActivity {
+
+    ImageButton placeimgButton;
+    TextView resNotificationTextView;
+    TextView placeTextView;
+    String selectedDate;
+
+    @Override
+        protected void onCreate(Bundle savedInstanceState) {
+            super.onCreate(savedInstanceState);
+            setContentView(R.layout.activity_place);
+
+        //Reservation Notification
+        resNotificationTextView = (TextView) this.findViewById(R.id.resNotificationText);
+
+        placeimgButton = (ImageButton) findViewById(R.id.cafeImgButton);
+        placeTextView = (TextView)findViewById(R.id.placeTextView);
+
+        placeimgButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                ApiInterface service = RetrofitInstance.getRetrofitInstance()
+                        .create(ApiInterface.class);
+
+                //get current date
+                long now = System.currentTimeMillis();
+                Date date = new Date(now);
+                SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+                selectedDate = sdf.format(date);
+
+                Call<ClassList> request = service.getClassList(selectedDate);
+
+                request.enqueue(new Callback<ClassList>() {
+                    @Override
+                    public void onResponse(Call<ClassList> call, Response<ClassList> response) {
+                        ClassList classList = response.body();
+
+                        Intent trendingClassActivity = new Intent(PlaceActivity.this,
+                                TrendingClassActivity.class);
+
+                        trendingClassActivity.putExtra("_classList", classList);
+                        trendingClassActivity.putExtra("_date", selectedDate);
+
+                        startActivity(trendingClassActivity);
+                    }
+
+                    @Override
+                    public void onFailure(Call<ClassList> call, Throwable t) {
+                        //TODO (woongjin) : how to deal with failure
+                    }
+                });
+            }
+        });
+
+        ApiInterface service = RetrofitInstance.getRetrofitInstance().create(ApiInterface.class);
+        Call<Reservation> request = service.getReservation("kahye5232@naver.com");
+        request.enqueue(new Callback<Reservation>() {
+            @Override
+            public void onResponse(Call<Reservation> call, Response<Reservation> response) {
+
+                //no reservation
+                if (response.code() == 203){
```

Comment:
If there's no reservation, the server should return back an empty response, and you can just check if the response is empty. I don't think 203 is a proper code for http responses to indicate errors.